provider/aws/aws_discover.go

```diff
@@ -142,6 +142,11 @@ func (p *Provider) Addrs(args map[string]string, l *log.Logger) ([]string, error
 		l.Printf("[INFO] discover-aws: Using static credentials provider")
 		staticCreds := credentials.NewStaticCredentialsProvider(accessKey, secretKey, sessionToken)
 		switch {
+		case addrType == "public_v4" || addrType == "private_v4":
+			cfg, err = config.LoadDefaultConfig(context.TODO(),
+				config.WithRegion(region),
+				config.WithCredentialsProvider(aws.NewCredentialsCache(staticCreds)),
+			)
```

Comment:
as this and the !found case are doing the same thing, could you please merge it with not found case by shifting the not found one above!